### PR TITLE
feat(generics): add metrics for database calls

### PR DESCRIPTION
**Pull request by NishantJoshi00**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Adding metrics for tracking database calls via prometheus

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by lsampras — APPROVED** [review]

File: crates/diesel_models/src/query/generics.rs [hunk 1]
```diff
@@ -22,25 +22,80 @@ use diesel::{
...
-    <T as QuerySource>::FromClause: QueryFragment<Pg>,
+    <T as QuerySource>::FromClause: QueryFragment<Pg> + Debug,
     <V as Insertable<T>>::Values: CanInsertInSingleQuery<Pg> + QueryFragment<Pg> + 'static,
     InsertStatement<T, <V as Insertable<T>>::Values>:
         AsQuery + LoadQuery<'static, PgConnection, R> + Send,
     R: Send + 'static,
 {
     let debug_values = format!("{values:?}");
+    let table_name = std::any::type_name::<T>().rsplit("::").nth(1);
+    logger::error!("YEE: {:?}", table_name);
```
Comment (SanchithHegde) [inline crates/diesel_models/src/query/generics.rs:90]: `YEE` 🙂

**Review by SanchithHegde — COMMENTED** [review]
Do we want to include metrics for the queries not using these generic functions as well? Some of the queries in process tracker, payments list, refunds list, etc.?

Other than that, looks good to me!

File: crates/diesel_models/src/query/generics.rs (hunk 1 above)
Comment (NishantJoshi00) [reply crates/diesel_models/src/query/generics.rs:90]: I'll change it to a more informative message.

Comment (NishantJoshi00) [reply crates/diesel_models/src/query/generics.rs:90]: Also, is there any better way to get table name, type name though simple i feel will incur some toll, does `Table` has something that we can use. Any thoughts?

**Review by SanchithHegde — DISMISSED** [review]

**Review by lsampras — DISMISSED** [review]

**Review by ashokkjag — APPROVED** [review]

**Review by ashokkjag — APPROVED** [review]

**Review by lsampras — APPROVED** [review]

**Pull request merged**